src/parquet/column/levels.h

```diff
@@ -95,23 +95,32 @@ class LevelDecoder {
  public:
   LevelDecoder() {}
 
-  // Initialize the LevelDecoder and return the number of bytes consumed
-  size_t Init(parquet::Encoding::type encoding, int16_t max_level,
+  // Initialize the LevelDecoder
+  void Init(int16_t max_level) {
+    bit_width_ = BitUtil::Log2(max_level + 1);
+    rle_decoder_.reset(new RleDecoder(NULL, 0, 0));
+    bit_packed_decoder_.reset(new BitReader(NULL, 0));
```

Comment:
If you move the initialization-from-null to Reset as discussed below then there won't be an issue (if the encoding were to change mid-stream then the then-null decoder would be initialized). But per my comment at the end BIT_PACKED may not be used anywhere at the moment out in the wild